x-pack/solutions/observability/plugins/observability/public/pages/alert_details/alert_ai_insight.tsx

```diff
@@ -75,12 +82,7 @@ export function AlertAiInsight({ alert }: { alert: AlertData }) {
     });
   };
 
-  const isObservabilityAgentEnabled = featureFlags.getBooleanValue(
-    OBSERVABILITY_AGENT_FEATURE_FLAG,
-    OBSERVABILITY_AGENT_FEATURE_FLAG_DEFAULT
-  );
-
-  if (!onechat || !isObservabilityAgentEnabled) {
+  if (!onechat || !hasAgentBuilderAccess) {
```

Comment:
Regarding [comment](https://github.com/elastic/kibana/pull/245299/commits/7fdfe7196686451f926cce67cba433d6859452f6#r2610501292), can we do something like:

`if (!onechat || !hasAgentBuilderAccess || !(chatExperience === AIChatExperience.Agent)) {
  return null;
}`